### PR TITLE
default CupertinoSliverNavigationBar's stretch to false

### LINKED ISSUE #74516: [CupertinoSliverRefreshControl] Refresh doesn't work and nothing happens on pull
## Steps to Reproduce

The bug is present on `1.26.0-1.0.pre-dev` as well as on `master` branch

1. Run `flutter create bug`.
2. Replace `MyHomePage` class with this code:
```dart
class MyHomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return CupertinoPageScaffold(
      child: CustomScrollView(
        physics: const BouncingScrollPhysics(
          parent: AlwaysScrollableScrollPhysics(),
        ),
        slivers: [
          CupertinoSliverNavigationBar(largeTitle: Text("Pull to refresh")),
          CupertinoSliverRefreshControl(onRefresh: () async {}),
          SliverList(
            delegate: SliverChildBuilderDelegate(
              (_, index) => Material(
                child: ListTile(title: Text("Item $index")),
              ),
              childCount: 20,
            ),
          ),
        ],
      ),
    );
  }
}
```

**Expected results:** 
This was run on `1.22.5-stable`

https://user-images.githubusercontent.com/23504276/105535296-ffe37280-5cee-11eb-99db-a8b52176db3c.mp4

**Actual results:** 
This was run on `1.26.0-1.0.pre-dev` and the same result was on `master` branch

https://user-images.githubusercontent.com/23504276/105535342-125dac00-5cef-11eb-8453-28f412631524.mp4

<details>
  <summary>Logs</summary>

```
```

```
Analyzing bug...
No issues found! (ran in 2.5s)
```

```
[✓] Flutter (Channel dev, 1.26.0-1.0.pre, on macOS 11.1 20C69 darwin-x64, locale
    en-GB)
    • Flutter version 1.26.0-1.0.pre at
      /Users/zengraf/.asdf/installs/flutter/1.26.0-1.0.pre-dev
    • Framework revision 63062a6443 (6 weeks ago), 2020-12-13 23:19:13 +0800
    • Engine revision 4797b06652
    • Dart version 2.12.0 (build 2.12.0-141.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/zengraf/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Users/zengraf/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.6953283/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/zengraf/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.6953283/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] Connected device (2 available)
    • iPhone 12 Pro Max (mobile) • 9E677C17-37B2-4E6B-BF9C-977D6E1B38D9 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • Chrome (web)               • chrome                               •
      web-javascript • Google Chrome 87.0.4280.141

• No issues found!
```

</details>

### PR DESCRIPTION
**Pull request by xster**
Default CupertinoSliverNavigationBar.stretch to false for now to fix https://github.com/flutter/flutter/issues/74516 until we create a more extensive system with https://github.com/flutter/flutter/issues/74652.

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/nav_bar.dart [hunk 1]
```diff
@@ -566,7 +566,7 @@ class CupertinoSliverNavigationBar extends StatefulWidget {
     this.padding,
     this.transitionBetweenRoutes = true,
     this.heroTag = _defaultHeroTag,
-    this.stretch = true,
+    this.stretch = false,
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/cupertino/nav_bar.dart:569]: the documentation of the class may need updating as well (currently it says the stretch property "Defaults to `true`").

**Review by LongCatIsLooong — APPROVED** [review]

File: packages/flutter/lib/src/cupertino/nav_bar.dart (hunk 1 above)
Comment (xster) [reply packages/flutter/lib/src/cupertino/nav_bar.dart:569]: I did here right https://github.com/flutter/flutter/pull/74669/files#diff-0f85f66e28e2ac684db897552ae5d0e9c4f7ddf326175e91f66beed3110900c2R666? Or was there somewhere else I missed?

Comment (LongCatIsLooong) [reply packages/flutter/lib/src/cupertino/nav_bar.dart:569]: Yeah it's mentioned here too:

https://github.com/flutter/flutter/blob/c8a1ea07249aa5ca750a00e82b89b18fadee4013/packages/flutter/lib/src/cupertino/nav_bar.dart#L541-L550

Comment (xster) [reply packages/flutter/lib/src/cupertino/nav_bar.dart:569]: thanks!

**Pull request merged**